aten/src/ATen/native/TensorShape.cpp

```diff
@@ -118,6 +118,7 @@ Tensor & _cat_out_cpu(Tensor& result, TensorList tensors, int64_t dim) {
         "unsupported operation: the input tensors cannot refer to any of the "
         "output memory locations. Found overlap in input tensor ", i);
   }
+  at::assert_no_internal_overlap(result);
```

Comment:
Add a test for `torch.cat(..., out=blah)`?

Reply:
The test was actually in the next PR up the stack. I've moved cat down into this PR because it already had some overlap checking already but otherwise moved most of the manual asserts up to the next PR where the corresponding test is.